toptal/picasso
# [FX-4508] Add package extraction script

[FX-4508](https://toptal-core.atlassian.net/browse/FX-4508)

### Description

**Review on the web only by commit!**

This PR introduces a script draft that extracts every component into a separate package.
The first commit contains the actual script, the rest is a result of script execution.

**Open the entire "files" tab on your own risk.**

### How to test


- [Temploy](https://picasso.toptal.net/fx-4508-what-is-the-cost-of-creating-component-by-package)
- Check the execution script in [first commit link] and see if it looks good
- CI might fail here and this is okay, it is not yet set up to execute tests on extracted packages
- Check out the code locally. Module resolution in extracted packages should work just fine, there should be no mismaches in tsconfig and actual packages

### Development checks


**Breaking change**


> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-4508]: https://toptal-core.atlassian.net/browse/FX-4508?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
